main.py

```diff
@@ -20,107 +21,131 @@ class DnsInfo(TypedDict):
     modified_on: str
 
 
-def list_zones() -> list[DnsInfo]:
-    '''List all zones associated with this cloudflare API key'''
+def list_zones() -> List[DnsInfo]:
+    """List all zones associated with this cloudflare API key"""
 
-    endpoint = 'zones'
+    endpoint = "zones"
     r = requests.get(api_url + endpoint, headers=headers)
 
     if r.status_code == 200:
-        return r.json()['result']
+        return r.json()["result"]
 
     else:
-        logging.error('HTTP: error listing domain')
+        logging.error("HTTP: error listing domain")
         logging.debug(r)
         raise RuntimeError("Listing DNS did not return 200")
 
 
-def get_zone_id_by_name(name: str) -> str:
-    '''Get a zone id associated with API key and domain name'''
+def get_zone_id_by_name(name: str) -> Optional[str]:
+    """Get a zone id associated with API key and domain name"""
 
     zone_data = list_zones()
     for zone in zone_data:
-        if zone['name'] == name:
-            return zone['id']
+        if zone["name"] == name:
+            return zone["id"]
+    return None
 
 
-def dns_info(zone_id: str) -> dict:
-    '''Get all dns information related to this zone id'''
+# NP - What is the structure of the returned dictionaory?
+def dns_info(zone_id: str) -> Dict[Any, Any]:
```

Comment:
as commented - no idea how to type this dictionary better - when I made it this my LSP started freaking out down below - obvi doesn't affect the code itself but as I've started typing much more fully it's been nice for the feedback from pylsp,mypy,flake8, etc.